mmdet/ops/nms/nms_wrapper.py

```diff
@@ -1,218 +1,22 @@
 
-    method_codes = {'linear': 1, 'gaussian': 2}
-    if method not in method_codes:
-        raise ValueError(f'Invalid method for SoftNMS: {method}')
-    results = nms_ext.soft_nms(dets_t, iou_threshold, method_codes[method], sigma,
-                               min_score)
-
-    new_dets = results[:, :5]
-    inds = results[:, 5]
-
-    if is_tensor:
-        return new_dets.to(
-            device=dets.device, dtype=dets.dtype), inds.to(
-                device=dets.device, dtype=torch.long)
-    else:
-        return new_dets.numpy().astype(dets.dtype), inds.numpy().astype(
-            np.int64)
-
-
-def batched_nms(bboxes, scores, inds, nms_cfg, class_agnostic=False):
-    """Performs non-maximum suppression in a batched fashion.
-
-    Modified from https://github.com/pytorch/vision/blob
-    /505cd6957711af790211896d32b40291bea1bc21/torchvision/ops/boxes.py#L39.
-    In order to perform NMS independently per class, we add an offset to all
-    the boxes. The offset is dependent only on the class idx, and is large
-    enough so that boxes from different classes do not overlap.
-
-    Arguments:
-        bboxes (torch.Tensor): bboxes in shape (N, 4).
-        scores (torch.Tensor): scores in shape (N, ).
-        inds (torch.Tensor): each index value correspond to a bbox cluster,
-            and NMS will not be applied between elements of different inds,
-            shape (N, ).
-        nms_cfg (dict): specify nms type and class_agnostic as well as other
-            parameters like iou_thr.
-        class_agnostic (bool): if true, nms is class agnostic,
-            i.e. IoU thresholding happens over all bboxes,
-            regardless of the predicted class
-
-    Returns:
-        tuple: kept bboxes and indice.
-    """
-    nms_cfg_ = nms_cfg.copy()
-    class_agnostic = nms_cfg_.pop('class_agnostic', class_agnostic)
-    if class_agnostic:
-        bboxes_for_nms = bboxes
-    else:
-        max_coordinate = bboxes.max()
-        offsets = inds.to(bboxes) * (max_coordinate + 1)
-
-        with no_nncf_trace():
-            # NB: this trick is required to make class-separate NMS using ONNX NMS operation;
-            #     the ONNX NMS operation supports another way of class separation (class-separate scores), but this is not used here.
-            # Note that `not_nncf_trace` is required here, since this trick causes accuracy degradation in case of int8 quantization:
-            #      if the output of the addition below is quantized, the maximal output value is about
-            #      ~ max_value_in_inds * max_coordinate,
-            #      usually this value is big, so after int8-quantization different small bounding
-            #      boxes may be squashed into the same bounding box, this may cause accuracy degradation.
-            # TODO: check if it is possible in this architecture use class-separate scores that are supported in ONNX NMS.
-            bboxes_for_nms = bboxes + offsets[:, None]
```

Comment:
@LeonidBeynenson This PR is to remove this `no_nncf_trace` context. Don't feel sorry to say what you think about it, please.

Reply:
Added no_nncf_trace to NMSop.forward and batched_nms.